### DISCUSSION TITLE
Can I use FormData in node-fetch using commonJS ?

### DISCUSSION THREAD

**Discussion by dante01yoon** [node-fetch/node-fetch#1597] [Q&A]
Hi, I'm using  
 "node-fetch": "^2.6.7"  version for support commonJS.

I want to use FormData via commonJS syntax which support in esm like
import { FormData } from "node-fetch"

is it possible?

**Comment by jimmywarting** [comment]
I think `FormData` was only implemented in v3...
there are ways you can load ESM modules async from cjs files if you need to. here are eg one solution if you can't switch to ESM just yet
```js
// commonjs file
import('node-fetch').then(mod => {
  const { default: fetch, FormData } = mod
  const fd = new FormData()
}) 
```
more over here: https://github.com/node-fetch/node-fetch/issues/1279

Alt you could use the built in fetch/FormData from NodeJS itself. they shipped it in NodeJS v18.x
See: https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#class-formdata

I just so happens to know that their `FormData` impl accepts 3th party blobs/files that are not instances of `globalThis.Blob` like `fetch-blob`. `fetch-blob` is pretty useful if you need to upload large files. if you want to see this landing in NodeJS core, give this issue https://github.com/nodejs/node/issues/37340, https://github.com/nodejs/node/issues/39015 a 👍

**Comment by ichenglin** [comment]
I believe `node-fetch` uses `form-data` package as its **dependencies**,
therefore instead of importing it from `node-fetch` you could instead **import it directly**.

Check out **[form-data](https://www.npmjs.com/package/form-data)** package on npm.js for further information.

```js
// import from form-data instead of node-fetch
import FormData from "form-data";

const form_data = new FormData();
form_data.append("name", "someone");
```

**Reply by jimmywarting** [reply]
We don't have a dependency on [form-data](https://www.npmjs.com/package/form-data)
but we do however support that one in `node-fetch@2.x` but we deprecated it and discourage others from using it as it's pretty far of from a standard spec compatible formdata implementation. 
a better alt would be to use:
- `formdata-polyfill` - also esm only... node-fetch depends on this package to decode payloads
- `formdata-node` - don't remember but i think this also recently switched to be esm only...? or plans to make it esm only.
- undici also have a built in FormData this is what NodeJS uses

**Reply by ichenglin** [reply]
He **mentioned** that the version of `node-fetch` he was using was `2.6.7`,
therefore `form-data` should **work** for him?

> Hi, I'm using
> "node-fetch": "^2.6.7" version for support commonJS.

**Reply by jimmywarting** [reply]
true...
